src/NzbDrone.Core/Indexers/IndexerStatusService.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NLog;
+using NzbDrone.Common.Cache;
+using NzbDrone.Core.Messaging.Events;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.ThingiProvider.Events;
+
+namespace NzbDrone.Core.Indexers
+{
+    public interface IIndexerStatusService
+    {
+        List<IndexerStatus> GetBlockedIndexers();
+        IndexerStatus GetIndexerStatus(int indexerId);
+        void ReportSuccess(int indexerId);
+        void ReportFailure(int indexerId, TimeSpan minimumBackOff = default(TimeSpan));
+
+        void UpdateRssSyncStatus(int indexerId, ReleaseInfo releaseInfo);
+    }
+
+    public class IndexerStatusService : IIndexerStatusService, IHandleAsync<ProviderDeletedEvent<IIndexer>>
+    {
+        private static readonly int[] EscalationBackOffPeriods = {
+                                                                     0,
+                                                                     5 * 60,
+                                                                     15 * 60,
+                                                                     30 * 60,
+                                                                     60 * 60,
+                                                                     3 * 60 * 60,
+                                                                     6 * 60 * 60,
+                                                                     12 * 60 * 60,
+                                                                     24 * 60 * 60
+                                                                 };
+        private static readonly int MaximumEscalationLevel = EscalationBackOffPeriods.Length - 1;
+
+        private static readonly object _syncRoot = new object();
+
+        private readonly IIndexerStatusRepository _indexerStatusRepository;
+        private readonly Logger _logger;
+
+        public IndexerStatusService(IIndexerStatusRepository indexerStatusRepository, Logger logger)
+        {
+            _indexerStatusRepository = indexerStatusRepository;
+            _logger = logger;
+        }
+
+        public List<IndexerStatus> GetBlockedIndexers()
+        {
+            return _indexerStatusRepository.All().Where(v => v.IsDisabled()).ToList();
+        }
+
+        public IndexerStatus GetIndexerStatus(int indexerId)
+        {
+            return _indexerStatusRepository.FindByIndexerId(indexerId) ?? new IndexerStatus { IndexerId = indexerId };
```

Comment:
I still think it should be responsibility of the test to simulate the context as it would be in production.